### ISSUE TITLE
[V3] Implement new Concept Exercise: sets

### ISSUE THREAD

**Issue by kristinaborn** [exercism/javascript#952]
This issue describes how to implement the `sets` concept exercise for the JavaScript track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This exercise aims to teach students how (and why) to use sets in JavaScript.

## Learning objectives

- Know how to add and remove elements from a `Set`
- Know how to check whether a value is in a `Set`
- Understand how `Set` checks equality
- Know how to iterate over `Set` keys, values, and entries
- Know how to convert between `Set` and `Array`
- Know how to use `Set` to remove duplicate elements from an array
- Know how `WeakSet` differs from `Set`
- Understand when `WeakSet` might be preferable to `Set`

## Out of scope

- Implementing common set operations like `union` and `difference`

## Concepts

- `sets`

## Prerequisites

- `arrays`
- `array-destructuring`
- `equality`

### After

- Use cases for `WeakSet`
- Lookup performance of `Set` vs. `Array`

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by kristinaborn** [comment]
I'm back from the dead and finally working on this 😅 

I think I'm going to opt not to address `WeakSet` in this exercise after all, unless folks feel strongly that I should. I believe that it's not used commonly enough outside of library code to be valuable to students.

**Comment by SleeplessByte** [comment]
No, I think we can have a separate exercise, if at all! I agree that people barely use it in regular every-day use.

**Comment by junedev** [comment]
This issue was resolved with #1093. Thanks @kristinaborn. :clap:
